libs/core/langchain_core/retrievers.py

```diff
@@ -22,8 +24,19 @@
         Callbacks,
     )
 
+
```

Comment:
Do we have examples of the metadata that folks want to add?

I would consider potentially adding some standard metadata the root level to force an identical schema between providers

```python
class DocumentResult(Serializable):
    document: Document
    extra: dict = Field(default_factory=dict)
    id: Optional[str]
    score: Optional[float]   
```

Reply:
In many search engines, what's returned is not the Document, but a hit snippet. A snippet is a chunk of text from the original document that matched.


```python
class Hit(Serializable):
    source: Optional[Document] # <-- The actual source of the document 
    snippet: ... # <-- Some type we don't know yet that might represent a chunk of the document
    extra: dict = Field(default_factory=dict)
    id: Optional[str]
    score: Optional[float]   
```